discv5/discv5-wire.md

```diff
@@ -263,14 +183,17 @@ PONG is the reply to PING.
 
 ### FINDNODE Request (0x03)
 
-    message-data = [request-id, distance]
+    message-data = [request-id, [distance₁, distance₂, ..., distanceₙ]]
     message-type = 0x03
-    distance     = the requested log2 distance, a positive integer
+    distanceₙ    = requested log2 distance, a positive integer
+
+FINDNODE queries for nodes at the given logarithmic distances from the recipient's node
+ID. When distance `0` is requested, the result set should contain the recipient's current
+record.
 
-FINDNODE queries for nodes at the given logarithmic distance from the recipient's node ID.
-The node IDs of all nodes in the response must have a shared prefix length of `distance`
-with the recipient's node ID. A request with distance `0` should return the recipient's
-current record as the only result.
+The recipient should create the result set by collecting nodes from its local node table
+according to the requested distances. Implementations should limit the number of nodes in
+the result set. The recommended result limit for FINDNODE queries is 16 nodes.
```

Comment:
May be make sense to settle some explicit limit on the total returned ENRs count for a single request (across all response packets)? The doc says '_should_ limit' but doesn't say what the limit _must_ be
I.e. I'm not sure if I should treat 100K ENRs returned as a remote node misbehavior or not.

Reply:
I've added something for this. I don't want to be to too specific about this yet, but it's definitely a good idea to limit it in the implementation. In go-ethereum, the limit is max 5 responses.